app/models/container_deployment.rb

```diff
@@ -6,5 +6,314 @@ class ContainerDeployment < ApplicationRecord
+    hash = {}
+    hash["ansible_config"] = ANSIBLE_CONFIG_LOCATION
+    hash["ansible_log_path"] = ANSIBLE_CONFIG_LOG
+    hash["ansible_inventory_path"] = ANSIBLE_CONFIG_INVENTORY_PATH
+    hash["ansible_ssh_user"] = ssh_auth.userid
+    hash["deployment"] = {}
+    hash["deployment"]["hosts"] = container_deployment_nodes.collect(&:to_ansible_config_format)
+    hash["deployment"]["roles"] = generate_roles
+    hash["version"] = version
+    hash["variant_version"] = kind.include?("origin") ? '1.2' : '3.2'
+    hash["variant"] = kind
+    # hash[:deployment][:roles][:master][:openshift_master_identity_providers] = []
+    # hash[:deployment][:roles][:master][:openshift_master_identity_providers] << identity_provider_auth.first.send(:to_ansible_config_format)
+    unless identity_provider_auth.first.htpassd_users.empty?
+      hash[:deployment][:roles][:master][:openshift_master_htpasswd_users] = {}
+      hash[:deployment][:roles][:master][:openshift_master_htpasswd_users] = identity_provider_auth.first.htpassd_users
+    end
+    hash.to_yaml.gsub("---\n", '')
+  end
+
+  def extract_public_ip_or_hostname(deployment_node)
+    return deployment_node.address if deployment_node.vm_id.nil?
+    vm = deployment_node.vm
+    hostname = vm.hostnames
+    if hostname.empty?
+      ips = vm.hardware.ipaddresses
+      ips.each do |ip|
+        return ip if public_ip? ip
+      end
+    end
+    hostname.first || ips.last
+  end
+
+  def public_ip?(ip)
+    require "ipaddr"
+    private = false
+    ip_addr = IPAddr.new(ip).to_i
+    ranges = ['10.0.0.0|10.255.255.255',
+              '172.16.0.0|172.31.255.255',
+              '192.168.0.0|192.168.255.255',
+              '169.254.0.0|169.254.255.255',
+              '127.0.0.0|127.255.255.255']
+    ranges.each do |range|
+      low, high = range.split("|")
+      low = IPAddr.new(low).to_i
+      high = IPAddr.new(high).to_i
+      private = true if (low..high).cover? ip_addr
+    end
+    return ip unless private
+  end
+
+  def generate_automation_params(params)
+    parameters = { # for all
+      :deployment_type   => kind.to_s,
+      :deployment_method => method_type,
+      :deployment_id     => id,
+      :provider_name     => params["provider_name"],
+      :containerized     => containerized,
+      :rhsub_sku         => rhsm_auth.rhsm_sku
+    }
+    if method_type.include? "provision"
+      parameters_provision = {
+        :ssh_public_key        => ssh_auth.public_key,
+        :provision_provider_id => deployed_on_ems.id,
+        :masters_provision     => {
+          :number_of_vms  => container_nodes_by_role("master").count,
+          :templateFields => {
+            "guid" => VmOrTemplate.find(params["masters_creation_template_id"]).guid, "name" => VmOrTemplate.find(params["masters_creation_template_id"]).name
+          },
+          :vmFields       => vm_fields("masters", params),
+          :requester      => {
+            'owner_email'      => 'temp@temp.com',
```

Comment:
VERY minor, but prefer `something@example.com` as example.com is reserved by IANA for testing/demo/example purposes: http://www.iana.org/domains/reserved